File: Cpp/1394_Find_Lucky_Integer_in_an_Array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        vector<int> frequency(501, 0);
        
        for(auto& num : arr){
            ++frequency[num];
        }
        
        for(int i = 500; i>=1; --i){
            if (i == frequency[i]) return i;
        }
        
        return -1;
    }
};

File: Kotlin/1394_Find_Lucky_Integer_in_an_Array.kt
class Solution {
    fun findLucky(arr: IntArray): Int {
        val frequency = IntArray(501)
        
        arr.forEach{ num ->
            ++frequency[num]
        }
        
        for(i in frequency.lastIndex downTo (1)) {
            if(frequency[i] == i) return i
        }
        
        return -1
    }
}